api/src/services/authentication.ts

```diff
@@ -409,17 +411,55 @@ export class AuthenticationService {
 			issuer: 'directus',
 		});
 
-		await this.knex('directus_sessions')
-			.update({
-				token: newRefreshToken,
-				expires: refreshTokenExpiration,
-			})
-			.where({ token: refreshToken });
+		if (options?.session) {
+			if (!record.session_next) {
+				// keep the old session active for a second
+				const SAFETY_TIMEOUT = 1000;
```

Comment:
What is a good timeout for this? 🤔

Reply:
From a safety-perspective, I think it should be acceptable to make this a bit higher (perhaps 5-10s?) to account for short delays of requests. IMO still a very small window for attacks.

Reply:
Yeah i agree, but still think it may be valuable top make configurable either way since its hard to tell how long of a timeout may be needed for huge databases/setups

Reply:
Yeah, sounds good 👍